Source/WTF/wtf/unix/UnixFileDescriptor.h

```diff
@@ -65,10 +67,20 @@ class UnixFileDescriptor {
         return *this;
     }
 
+    UnixFileDescriptor& operator=(const UnixFileDescriptor& o)
+    {
+        if (&o == this)
+            return *this;
+
+        this->~UnixFileDescriptor();
+        new (this) UnixFileDescriptor(o);
+        return *this;
+    }
+
     ~UnixFileDescriptor()
     {
         if (m_value >= 0)
-            closeWithRetry(m_value);
+            closeWithRetry(std::exchange(m_value, -1));
```

Comment:
FWIW, nothing can touch the `m_value` after destructor, so the `exchange` part is dead code.

Reply:
Right, Zan added that because I tricked him into thinking it was necessary... my bad.

Reply:
I'm going to unset merge-queue one more time just to make sure Zan has seen this. Sorry. :P

Reply:
I think it's fine to set it back to -1, it's not costly.